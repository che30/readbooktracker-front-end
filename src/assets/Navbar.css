.Logout__logo{
  padding-right: 2%;
}
.Navcontent {
  padding-left: 44%;
}
nav {
  font-size: 1.8rem;
  justify-content: space-between;
  align-items: center;
  padding: 5px 0 5px 0;
}
.sign__out__Btn{
  color: white;
  opacity: 1;
  background-color: #3498db;
  border: none;
}
@media(max-width: 567px) {
  .Navcontent {
    padding-left: 25%;
  }
}