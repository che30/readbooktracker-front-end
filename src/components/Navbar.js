import React from 'react';
import Proptypes from 'prop-types';
import '../assets/Navbar.css';

const Navbar = ({ Navcontent }) => {
  const handleClick = () => {
    localStorage.setItem('auth_token', null);
    window.location.href = '/Login';
  };
  return (
    <>
      <nav className="bg_color_Pantone w-100 text-white d-flex font-helvetica-bold">
        <div className="Navcontent">{Navcontent}</div>
        <div className="Logout__logo">
          <button
            className="sign__out__Btn"
            onClick={handleClick}
            type="button"
            aria-label="Logout"
          >
            <i className="fa fa-power-off" />
          </button>
        </div>
      </nav>
    </>
  );
};
Navbar.propTypes = {
  Navcontent: Proptypes.string.isRequired,
};
export default Navbar;
