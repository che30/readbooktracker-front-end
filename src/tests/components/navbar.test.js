import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Navbar from '../../components/Navbar';

Enzyme.configure({ adapter: new Adapter() });
describe('Navabr', () => {
  it('wraps content in a nav with .Navcontent class', () => {
    const wrapper = shallow(<Navbar Navcontent="Dashboard" />);
    expect(wrapper.find('.Navcontent').length).toEqual(1);
    expect(wrapper.find('.Navcontent').text()).toBe('Dashboard');
  });
  it('displays the current Navbar', () => {
    const wrapper = shallow(<Navbar Navcontent="Dashboard" />);
    expect(wrapper.find('.Navcontent').text()).toBe('Dashboard');
  });
  it('Logs out a user', () => {
    const wrapper = shallow(<Navbar Navcontent="Dashboard" />);
    wrapper.find('.sign__out__Btn').simulate('click');
    expect(wrapper.find('nav').hasClass('Navcontent')).toBeFalsy();
  });
  it('renders properly', () => {
    const tree = renderer
      .create(<Navbar Navcontent="Dashboard" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
