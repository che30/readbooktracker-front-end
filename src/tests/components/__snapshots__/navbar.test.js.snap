// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navabr renders properly 1`] = `
<nav
  className="bg_color_Pantone w-100 text-white d-flex font-helvetica-bold"
>
  <div
    className="Navcontent"
  >
    Dashboard
  </div>
  <div
    className="Logout__logo"
  >
    <button
      aria-label="Logout"
      className="sign__out__Btn"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-power-off"
      />
    </button>
  </div>
</nav>
`;
